src/main/java/butler/util/Parser.java

```diff
@@ -0,0 +1,137 @@
+package butler.util;
+
+import butler.ButlerInputException;
+import butler.command.Command;
+import butler.command.ByeCommand;
+import butler.command.DeadlineCommand;
+import butler.command.DeleteCommand;
+import butler.command.EventCommand;
+import butler.command.FindCommand;
+import butler.command.InvalidCommand;
+import butler.command.ListCommand;
+import butler.command.MarkCommand;
+import butler.command.TodoCommand;
+import butler.command.UnmarkCommand;
+
+public class Parser {
+
+    /**
+     * Parses user input and check if it matches any predefined commands.
+     * Predefined commands are:
+     * bye
+     * list
+     * mark
+     * unmark
+     * todo
+     * deadline
+     * event
+     * find
+     * delete
+     *
+     * @param input User input.
+     * @return Command Relevant command to execute given the input.
+     * @throws ButlerInputException This exception is thrown via the InvalidCommand if input matches no given commands.
+     */
+    public static Command parse(String input) throws ButlerInputException {
+        String[] stringArray = input.split(" ");
+        String firstWord = stringArray[0];
+        if (firstWord.equals("bye")) {
+            return new ByeCommand();
+        } else if (firstWord.equals("list") && stringArray.length == 1) {
+            return new ListCommand();
+        } else if (firstWord.equals("mark")) {
+            if (stringArray.length != 2) {
+                throw new ButlerInputException("Error. Sorry Master, " +
+                        "but a mark or unmark command must be followed by one Integer.\n");
+            }
+            try {
+                return new MarkCommand(Integer.parseInt(stringArray[1]) - 1);
+            } catch (NumberFormatException exception) {
+                throw new ButlerInputException("Error. Sorry Master, " +
+                       "but a mark or unmark command must be followed by one Integer.\n");
+            }
```

Comment:
Is there a reason why you chose to handle the case where `stringArray.length != 2` explicitly instead of leaving it to the try-catch block?

Reply:
Hi Justine! The reason I chose to handle the case of stringArray.length != 2 here is because it is the job of the parser  to check for input. The try catch block should be reserved for sections of code that throws exceptions further down the function call stack. Even if I did place the if statement in the try catch block, the parser would not know what to do with the exception message and would have to rethrow it anyway, so I just placed it outside the try catch block.